### PR TITLE
EMS API and UI code

### PR DESCRIPTION
**Pull request by surekhag**

### CODE REVIEW

File: client/src/layouts/Login/Login.js [hunk 1]
```diff
@@ -49,27 +42,25 @@ const styles = {
 const useStyles = makeStyles(styles)
 
 function Login(props) {
```
Comment (bhushan-oe) [inline client/src/layouts/Login/Login.js]: use arrow function syntax

File: server/app/api/controllers/projects.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
+const projectModel = require("../models/projects");
+// const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
```
Comment (bhushan-oe) [inline server/app/api/controllers/projects.js]: unused jwt

File: server/app/api/controllers/projects.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        else
+          res.json({
+            status: "success",
+            message: "Project added successfully!!!",            
+          });
+      }
+    );
+  },
+  getProject: function(req, res, next) {
+    console.log(req.user);
```
Comment (bhushan-oe) [inline server/app/api/controllers/projects.js]: remove log

File: server/app/api/controllers/projects.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        res.json({
+          status: "success",
+          message: "Project found!!!",
+          data: users
+        });
+      }
+    });
+  },
+  getAll: function(req, res, next) {
+    console.log(req.user);
```
Comment (bhushan-oe) [inline server/app/api/controllers/projects.js]: remove log

File: server/app/api/controllers/projects.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+  delete: function(req, res, next) {
+    projectModel.findOneAndUpdate({ title: req.params.title }, 
+      {
+         status : "Inactive"
+      },
+      {upsert: true}, 
+      function(err, userInfo) {      
+      if (err) {
+        console.log("in err");
```
Comment (bhushan-oe) [inline server/app/api/controllers/projects.js:84]: remove log

File: server/app/api/controllers/projects.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
+const projectModel = require("../models/projects");
+// const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const config = require("../../../config");
```
Comment (bhushan-oe) [inline server/app/api/controllers/projects.js]: config is not used

File: server/app/api/controllers/users.js [hunk 7]
```diff
@@ -6,10 +6,32 @@ module.exports = {
   create: function(req, res, next) {
     userModel.create(
       {
-        userName: req.body.userName,
-        email: req.body.email,
-        password: req.body.password,
-        userRole: req.body.userRole
+          employee_id : req.body.employee_id ,
+          email: req.body.email ,
```
Comment (bhushan-oe) [inline server/app/api/controllers/users.js:10]: what will happen if field is not present in request body

File: server/app/api/controllers/users.js [hunk 8]
```diff
@@ -71,5 +92,44 @@ module.exports = {
         });
       }
     });
+  },
+  update: function(req, res, next) {    
+    userModel.findOneAndUpdate({ userName: req.params.userName }, 
+      {
+          $set: req.body 
+      },
+      {upsert: true},
```
Comment (bhushan-oe) [inline server/app/api/controllers/users.js]: why do we need upsert?

File: server/routes/projects.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+const express = require("express");
+const router = express.Router();
+const projectController = require("../app/api/controllers/projects");
+
+router.get("/", projectController.getAll);
+router.post("/add", projectController.create);
+router.get("/:title", projectController.getProject);
```
Comment (bhushan-oe) [inline server/routes/projects.js]: keep it on id , not on title

File: server/routes/users.js [hunk 10]
```diff
@@ -5,5 +5,8 @@ const userController = require("../app/api/controllers/users");
 router.get("/", userController.getAll);
 router.post("/register", userController.create);
 router.post("/login", userController.authenticate);
+router.post("/add", userController.create);
+router.put("/:userName", userController.update);
```
Comment (bhushan-oe) [inline server/routes/users.js]: keep it on id, not on username

**Review by bhushan-oe — CHANGES_REQUESTED** [review]

File: server/app/api/controllers/users.js (hunk 7 above)
Comment (surekhag) [reply server/app/api/controllers/users.js:10]: Validation is displayed for the mandatory field.

File: server/app/api/controllers/users.js (hunk 8 above)
Comment (surekhag) [reply server/app/api/controllers/users.js]: This is part of the syntax for findOneAndUpdate where upsert : upsert: bool - creates the object if it doesn't exist. defaults to false.  I have removed Upsert now

File: server/routes/projects.js (hunk 9 above)
Comment (surekhag) [reply server/routes/projects.js]: Done

File: server/routes/users.js (hunk 10 above)
Comment (surekhag) [reply server/routes/users.js]: Done

File: server/app/api/controllers/projects.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+              res.json({
+                status: "success",
+                message: "Project updated successfully!!!",                
+              });
+            }
+        });
+  },
+
+  delete: function(req, res, next) {
+    projectModel.findOneAndUpdate({ _id: req.params._id },
```
Comment (bhushan-oe) [inline server/app/api/controllers/projects.js]: request param can be "id" instead "_id"

File: server/app/api/controllers/users.js (hunk 7 above)
Comment (bhushan-oe) [reply server/app/api/controllers/users.js:10]: ok

File: server/app/api/controllers/users.js [hunk 12]
```diff
@@ -71,5 +90,42 @@ module.exports = {
...
+              res.json({
+                status: "success",
+                message: "User Info  updated successfully!!!",                
+              });
+            }
+        });
+  },
+
+  delete: function(req, res, next) {
+    userModel.findOneAndUpdate({ _id: req.params._id },
```
Comment (bhushan-oe) [inline server/app/api/controllers/users.js]: change to "id"

File: server/app/api/models/leaves_application.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
+
+
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+const Leaves_application_Schema = new Schema({
```
Comment (bhushan-oe) [inline server/app/api/models/leaves_application.js]: keep case consistent

File: server/app/api/models/leaves_application.js [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+        required: true,        
+      },
+    last_updated_by  : {
+        type: String,
+        trim: true,
+        required: true,        
+      },
+});
+
+module.exports = mongoose.model("PeerReview", Leaves_application_Schema);
```
Comment (bhushan-oe) [inline server/app/api/models/leaves_application.js]: model name is not matching

File: server/app/api/models/leaves_application_summary.js [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+        required: true,        
+      },
+    updated_by : {
+        type: String,
+        trim: true,
+        required: true,        
+      },
+});
+
+module.exports = mongoose.model("PeerReview", Leaves_application_summary_Schema);
```
Comment (bhushan-oe) [inline server/app/api/models/leaves_application_summary.js]: model name not matching

File: server/app/api/models/leaves_credit_summary.js [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+        required: true,        
+      },
+    updated_by  : {
+        type: String,
+        trim: true,
+        required: true,        
+      },
+});
+
+module.exports = mongoose.model("PeerReview", Leaves_credit_summary_Schema);
```
Comment (bhushan-oe) [inline server/app/api/models/leaves_credit_summary.js]: model name is not matching

File: server/app/api/models/project_allocation.js [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+        required: true,        
+      },
+    last_updated_by  : {
+        type: String,
+        trim: true,
+        required: true,        
+      },
+});
+
+module.exports = mongoose.model("PeerReview", Project_allocation_Schema);
```
Comment (bhushan-oe) [inline server/app/api/models/project_allocation.js]: model name

File: server/app/api/models/work_outside_office_log.js [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+        required: true,        
+      },
+    status : {
+        type: String,
+        trim: true,
+        required: true,        
+      },
+});
+
+module.exports = mongoose.model("PeerReview", Work_outside_office_log_Schema);
```
Comment (bhushan-oe) [inline server/app/api/models/work_outside_office_log.js]: All schema variables can be camelCase e.g. `Work_outside_office_log_Schema` can be `workOutsiteOfficeLogSchema`

**Review by bhushan-oe — CHANGES_REQUESTED** [review]

**Pull request merged**